storage_service/locations/models/space.py

```diff
@@ -143,16 +143,31 @@ def get_child_space(self):
         return protocol_space
 
     def browse(self, path, *args, **kwargs):
-        """ Returns {'directories': [directory], 'entries': [entries]} at path.
-
-        `path` is a full path in this space.
-
-        'directories' in the return dict is the name of all the directories
-            located at that path
-        'entries' in the return dict is the name of any file (directory or other)
-            located at that path
-
-        If not implemented in the child space, looks locally.
+        """
+        Return information about the objects (files, directories) at `path`.
+
+        Attempts to call the child space's implementation.  If not found, falls
+        back to looking for the path locally.
+
+        Returns a dictionary with keys 'entries', 'directories' and 'properties'.
+
+        'entries' is a list of strings, one for each entry in that directory, both file-like and folder-like.
+        'directories' is a list of strings for each folder-like entry. Each entry should also be listed in 'entries'.
+        'properties' is a dictionary that may contain additional information for the entries.  Keys are the entry name found in 'entries', values are a dictionary containing extra information. 'properties' may not contain all values from 'entries'.
```

Comment:
Sure, that'd help! I'm looking at this as a consumer of the data later, wanting to know what kind of data is actually going to be available - otherwise I wouldn't be sure what I could do with it.

If we could document in the spaces what properties they can provide that would also be helpful!